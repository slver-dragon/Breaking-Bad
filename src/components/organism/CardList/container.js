import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import CardList from "./component";
import Paginator from "../../molecule/Pagintor";
import Button from "../../atom/Button";
import { loadPersons } from "../../../store/actions/getPersonData";
import {
  changeCatalogFormat,
  changeElementCount,
  setCurrentPage,
} from "../../../store/actions/services";
import style from "./style.module.scss";
import cardsListOn from "../../../assets/img/listOn.png";
import cardsListOff from "../../../assets/img/listOff.png";
import cardsGridOn from "../../../assets/img/gridOn.png";
import cardsGridOff from "../../../assets/img/gridOff.png";

const PersonListContainer = () => {
  const dispatch = useDispatch();
  const characters = useSelector((state) => state.personData.characters);
  const cardsMaxCount = useSelector(
    (state) => state.personData.charactersMaxLength
  );
  const isLoading = useSelector((state) => state.personData.isLoad);
  const errorValue = useSelector((state) => state.personData.errorValue);
  const isCardsList = useSelector((state) => state.services.isCardsList);
  const currentPage = useSelector((state) => state.services.currentPage);
  const changeCount = (value) => dispatch(changeElementCount(value));
  const changePage = (value) => dispatch(setCurrentPage(value));
  const pageElementCount = useSelector(
    (state) => state.services.pageElementCount
  );
  const [cardsListImage, cardsGridImage] = isCardsList
    ? [cardsListOn, cardsGridOff]
    : [cardsListOff, cardsGridOn];
  const titleMenu = isCardsList ? (
    <div>
      <div className={style.wrapper}>
        <div className={style.pictureCap} alt="" />
        <div className={style.titleMenu}>
          <span className={style.status}>Статус</span>
          <span className={style.name}>Имя</span>
          <span className={style.birthday}>Дата рождения</span>
        </div>
      </div>
    </div>
  ) : (
    ""
  );
  const divider = isCardsList ? <div className={style.divider} /> : "";
  useEffect(
    () => {
      dispatch(loadPersons(0, "all", true));
    },
    // eslint-disable-next-line
    []
  );
  useEffect(
    () => {
      dispatch(
        loadPersons(
          (currentPage - 1) * pageElementCount,
          pageElementCount,
          true
        )
      );
    },
    // eslint-disable-next-line
    [currentPage, pageElementCount, isCardsList]
  );
  useEffect(() => {
    if (isCardsList && window.innerWidth <= 600) {
      dispatch(changeCatalogFormat(false));
    }
  });

  return (
    <div className={style.main}>
      <div className={style.changer}>
        <span>Каталог</span>
        <div className={style.buttonBlock}>
          <span onClick={() => dispatch(changeCatalogFormat(true))}>
            <Button img={cardsListImage} text={""} />
          </span>
          <span onClick={() => dispatch(changeCatalogFormat(false))}>
            <Button img={cardsGridImage} text={""} />
          </span>
        </div>
      </div>
      {titleMenu}
      <CardList
        id="CardList"
        characters={characters}
        isLoading={isLoading}
        errorValue={errorValue}
        isCardsList={isCardsList}
      />
      {divider}
      <Paginator
        currentPage={currentPage}
        pageElementCount={pageElementCount}
        cardsMaxCount={cardsMaxCount}
        changeCount={changeCount}
        changePage={changePage}
      />
    </div>
  );
};

export const container = PersonListContainer;
