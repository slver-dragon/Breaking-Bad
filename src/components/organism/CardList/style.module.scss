@import "../../../assets/styles/style";
.container {
  padding: 0;
  margin: 0;
  .main {
    display: flex;
    flex-direction: column;
    min-height: 60vh;
    @include mainContainer;
    .changer {
      display: inline-flex;
      justify-content: space-between;
      padding: 0.5rem 0;
      span {
        @include personPageBigFont;
      }
      .buttonBlock {
        @include buttonBlock;
      }
    }
    .wrapper {
      display: flex;
      justify-content: space-between;
      .pictureCap {
        padding: 0 2rem;
      }
      .titleMenu {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        align-items: center;
        justify-content: space-around;
        margin: 1.5rem 0 2rem;
        @include mediumFont;
        font-size: 0.875rem;
        flex: 1 1 auto;
        span {
          margin: 0 auto;
        }
      }
    }
    .wrapperGrid {
      display: grid;
      column-gap: 2rem;
      justify-content: space-between;
      @include cardListWrapperGrid;
    }
    .wrapperList {
      display: flex;
      flex-direction: column;
    }
    .divider {
      border-top: 0.0625rem solid $gray;
    }
  }
  .paginator {
    @include mainContainer;
  }
}
