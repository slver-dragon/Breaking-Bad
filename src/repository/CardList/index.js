import axios from "axios";

class CardList {
  createCardList = async () => {
    let result = {
      value: null,
      error: null,
    };
    try {
      const response = await axios.get(
        "https://www.breakingbadapi.com/api/characters?limit=10"
      );
      result.value = response.data;
    } catch (error) {
      result.error = error.message;
    }
    return result;
  };
}

export default new CardList();
