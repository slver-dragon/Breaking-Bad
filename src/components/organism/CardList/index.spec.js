import React from "react";
import { shallow } from "enzyme";
import CardList from "./component";
// import PersonListContainer from ".";

describe("Testing organism CardList:", () => {
  let component = null;
//   const onBtnClick = jest.fn();
//   const onClickEnter = jest.fn();
  const setComponent = (props) => shallow(<CardList {...props} />);
//   const setContainer = (props) => shallow(<PersonListContainer {...props} />);

  test("component is rendering", () => {
    component = setComponent();
    expect(component).toMatchSnapshot();
  });

//   test("container is rendering", () => {
//     component = setContainer();
//     expect(component).toMatchSnapshot();
//   });
});