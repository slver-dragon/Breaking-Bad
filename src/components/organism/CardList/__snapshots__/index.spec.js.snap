// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing organism CardList: component is rendering as grid 1`] = `
<div>
  <div
    className="wrapperGrid"
  >
    <PersonCard
      id={1}
      key="1"
    />
    <PersonCard
      id={2}
      key="2"
    />
  </div>
</div>
`;

exports[`Testing organism CardList: component is rendering as list 1`] = `
<div>
  <div
    className="wrapperList"
  >
    <PersonCard
      id={1}
      key="1"
    />
    <PersonCard
      id={2}
      key="2"
    />
  </div>
</div>
`;
